engine/apps/mattermost/tests/test_backend.py

```diff
@@ -0,0 +1,91 @@
+import pytest
+from rest_framework import serializers
+
+from apps.mattermost.backend import MattermostBackend
+from apps.user_management.models import User
+
+
+@pytest.mark.django_db
+def test_unlink_user(make_organization_and_user, make_mattermost_user):
+    _, user = make_organization_and_user()
+    make_mattermost_user(user=user)
+    backend = MattermostBackend()
+    backend.unlink_user(user)
+    user.refresh_from_db()
+
+    with pytest.raises(User.mattermost_user_identity.RelatedObjectDoesNotExist):
+        user.mattermost_user_identity
+
+
+@pytest.mark.django_db
+def test_serialize_user(make_organization_and_user, make_mattermost_user):
+    _, user = make_organization_and_user()
+    mattermost_user = make_mattermost_user(user=user)
+    data = MattermostBackend().serialize_user(user)
+    assert data["mattermost_user_id"] == mattermost_user.mattermost_user_id
+    assert data["username"] == mattermost_user.username
+
+
+@pytest.mark.django_db
+def test_serialize_user_not_found(
+    make_organization_and_user,
+):
+    _, user = make_organization_and_user()
+    data = MattermostBackend().serialize_user(user)
+    assert data is None
+
+
+@pytest.mark.django_db
+def test_validate_channel_filter_data(
+    make_organization,
+    make_mattermost_channel,
+):
+    organization = make_organization()
+    channel = make_mattermost_channel(organization=organization, is_default_channel=True)
+    input_data = {"channel": channel.public_primary_key, "enabled": True}
+    data = MattermostBackend().validate_channel_filter_data(organization=organization, data=input_data)
+    assert data["channel"] == channel.public_primary_key
+    assert data["enabled"]
+
+
+@pytest.mark.django_db
+def test_validate_channel_filter_data_update_only_channel(
+    make_organization,
+    make_mattermost_channel,
+):
+    organization = make_organization()
+    channel = make_mattermost_channel(organization=organization, is_default_channel=True)
+    input_data = {"channel": channel.public_primary_key}
+    data = MattermostBackend().validate_channel_filter_data(organization=organization, data=input_data)
+    assert data["channel"] == channel.public_primary_key
+    assert "enabled" not in data
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize(
+    "input_data,expected_data",
+    [
+        ({"enabled": True}, {"enabled": True}),
+        ({"enabled": False}, {"enabled": False}),
+        ({"channel": None, "enabled": True}, {"channel": None, "enabled": True}),
+        ({"channel": None}, {"channel": None}),
```

Comment:
`input_data` and `expected_data` are always the same? I think you are missing some case to cover when you make changes to input during validation? (e.g. passing `"enabled": 1`? some of the other exceptional cases?)

Reply:
Yeah the idea of this test case is to cover the toggle option flow and the option to clear the channel once it is selected using the `X` on the dropdown. For other exception cases I have added separate tests [ref](https://github.com/grafana/oncall/pull/5347/files#diff-b9c01c665a1c5d4873783e707a541cab535839629a294638c1cde5d894ab8ca3R88). Do you see anyother case that needs to be covered in the tests ?